metal/resource_metal_virtual_circuit.go

```diff
@@ -78,13 +78,17 @@ func resourceMetalVirtualCircuitCreate(d *schema.ResourceData, meta interface{})
 	client := meta.(*packngo.Client)
 	vncr := packngo.VCCreateRequest{
 		VirtualNetworkID: d.Get("vlan_id").(string),
-		NniVLAN:          d.Get("nni_vlan").(int),
 		Name:             d.Get("name").(string),
 	}
+
 	connId := d.Get("connection_id").(string)
 	portId := d.Get("port_id").(string)
 	projectId := d.Get("project_id").(string)
 
+	if nniVlan, ok := d.GetOk("nni_vlan"); ok {
+		vncr.NniVLAN = nniVlan.(int)
+	}
+
```

Comment:
Shall we do a sanity check for when user set nni_vlan but connection mode is not `tunnel`?